### PR TITLE
fix: phase 1 fixes

### PR DESCRIPTION
**Pull request by laszlokr**

### CODE REVIEW

**Comment by janikvonrotz** [comment]
- Select records in the asset -> Maybe a new selection field `name_ids` with a computed function that returns all records with (id, displayname). When selecting a value the id is written to `res` field
- add the project_id fields
- define the default value for mgmt_audit_project_id in a data.xml
- Move the project demo data from demo.xml to data.xml
- Add the color widget to select widget

File: mgmt_audit_project/models/nonconformity.py [hunk 1]
```diff
@@ -10,3 +10,12 @@ class MgmtNonconformity(models.Model):
 
     name = fields.Char(required=True)
     task_ids = fields.One2many("project.task", "nonconformity_id")
+
+    @api.model
+    def _get_default_project_id(self):
+        project = self.env.ref("mgmt_audit_project.mgmt_project", raise_if_not_found=False)
+        if project:
+            return project.id
+        return self.env['project.project'].search([], limit=1).id
```
Comment (janikvonrotz) [inline mgmt_audit_project/models/nonconformity.py]: Return the first project is wrong. It should return `False`.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_audit_project/models/nonconformity.py (hunk 1 above)
Comment (laszlokr) [reply mgmt_audit_project/models/nonconformity.py]: @janikvonrotz like this?
```
        def _get_default_project_id(self):
        project = self.env.ref("mgmt_audit_project.mgmt_project", raise_if_not_found=false)
        if project:
            return project.id
        return false
```

File: mgmt_audit_project/views/project_task_views.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <field name="inherit_id" ref="project.view_task_form2" />
         <field name="arch" type="xml">
             <field name="name" position="after">
-                <field name="name" />
+                <!-- <field name="name" /> -->
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/project_task_views.xml]: Why is this commented? Can you remove it?

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

File: mgmt_audit_project/views/nonconformity_views.xml [hunk 3]
```diff
@@ -6,7 +6,11 @@
         <field name="inherit_id" ref="mgmt_audit.nonconformity_form_view" />
         <field name="arch" type="xml">
             <field name="type" position="after">
```
Comment (janikvonrotz) [inline mgmt_audit_project/views/nonconformity_views.xml]: The indentation seems wrong here.

**Review by janikvonrotz — CHANGES_REQUESTED** [review]

**Comment by laszlokr** [comment]
@janikvonrotz I have an issue with loading demo data for `mgmt_asset` which I am not sure how to resolve, could you please help. 
```
TypeError: 'NoneType' object is not iterable

    raise ParseError('while parsing %s:%s, somewhere inside\n%s' % (
odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:12, somewhere inside<record id="demo_asset_server" model="mgmt.asset">
        <field name="name">Servers</field>
        <field name="model_name">maintenance.equipment</field>
        <field name="res_id" eval="ref('mgmt_asset.demo_inventory_server')"/>
    </record>
```

**Comment by janikvonrotz** [comment]
@laszlokr I just installed the module and did not have any problems. Please reset the db.

**Comment by janikvonrotz** [comment]
@laszlokr I will now start fixing the PR. If you have anything the working directory please commit.

**Comment by laszlokr** [comment]
> @laszlokr I just installed the module and did not have any problems. Please reset the db.

@janikvonrotz I have reset the database like 5 times while trying to fix this error, I don't understand what's the problem here

**Comment by laszlokr** [comment]
> @laszlokr I will now start fixing the PR. If you have anything the working directory please commit.

@janikvonrotz I have nothing important to commit, it's better if you can just fix it. Sorry that I could not bring it to completion on my own

**Comment by janikvonrotz** [comment]
@laszlokr I can now reproduce the issue.

**Comment by janikvonrotz** [comment]
@laszlokr I have a lot of linter issues. Please before every commit, run the linter.

**Comment by janikvonrotz** [comment]
@laszlokr The indentation in the xml files is wrong in many cases.

**Comment by laszlokr** [comment]
> @laszlokr I have a lot of linter issues. Please before every commit, run the linter.

@janikvonrotz I do run the linter before every commit, but I can see there is some issue with my setup
`[Errno 2] No such file or directory: '/home/laszlo/.cache/pre-commit/repohkcoj8rq/py_env-python3/bin/ruff'`
Running `task init` or `pip install ruff` does not solve this. As I understand the problem is due to nix-shell because the linter caught errors on the other system. I will investigate what can be done

**Pull request merged**